### PR TITLE
[SCL-148] supply data for species report chart

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[Support time range in species report](https://trello.com/c/wQF8RVL6/148-support-time-range-in-species-report)

File: src/api/reports/species_report.py [hunk 1]
```diff
@@ -28,8 +34,28 @@ def open_new_template(template):
...
+
+    if is_sequence(data[0]):
+        w = len(data[0])
+    else:
+        w = 1
+
+    start_letter = get_column_letter(p1)
+    end_letter = get_column_letter(p1 + w - 1)
+
+    return rows_from_range(f"{start_letter}{p0}:{end_letter}{p0 + h - 1}")
```
Comment (saanobhaai) [inline src/api/reports/species_report.py:53]: `h` is undefined

File: src/api/reports/species_report.py [hunk 2]
```diff
@@ -40,15 +66,21 @@ def generate(data):
     workbook = load_workbook(filename=report_path)
     sheet = workbook.active
 
-    for key, value in data.items():
-        coord = DATA_CELL_MAP.get(key)
-        if not coord:
-            continue
+    for sheet_name, sheet_data in data.items():
+        sheet = workbook[sheet_name]
```
Comment (saanobhaai) [inline src/api/reports/species_report.py:73]: Protect against `sheet_name` not in `workbook` `KeyError`?

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Just a couple questions, plus one I'll ask you about in Slack

**Pull request merged**